### PR TITLE
Usb modeswitch [WIP]

### PR DESCRIPTION
**Pull request by spinus**
Hi,

as there is few issues with USB modems in NixOS I'm trying to fix it by adding udev rules (usb-modeswitch-data). 

As you can see I added missing part of usb-modeswitch package which is udev rules from usb-modeswitch-data package.
I'm not sure how to expose those udev rules to the system. After I install the package (as root), udev rules are not in /etc/udev/rules.d nor /lib/udev/rules.d. Do I need to add nixos "service" or "module" to expose that rules from a package?

Also, maybe will be good to add usb-modeswitch and usb-modeswitch-data as modemmanager dependency so people which install modemmanager have their usb-modems enabled by default?

### CODE REVIEW

File: pkgs/development/tools/misc/usb-modeswitch-data/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+  # make clean: we always build from source. It should be necessary on x86_64 only
+  prePatch = ''
+    sed -i 's@usb_modeswitch@${usb-modeswitch}/bin/usb_modeswitch@g' 40-usb_modeswitch.rules
+    sed -i "1 i\DESTDIR=$out" Makefile
+  '';
+
+  buildInputs = [ pkgconfig libusb1 usb-modeswitch ];
+
+  meta = {
```
Comment (fpletz) [inline pkgs/development/tools/misc/usb-modeswitch-data/default.nix]: The description is missing.

**Comment by fpletz** [comment]
Yes, there is the option `services.udev.packages` for exactly that. For example, you could create a NixOS module that adds an option in the `hardware` hierarchy to enable usb-modeswitch. This would append the usb-modeswitch package to `services.udev.packages` which would symlink all udev rules from `$out/etc/udev/rules.d` into the actualy path in your NixOS system.

Looks good otherwise. :+1:

**Comment by spinus** [comment]
@fpletz thank you, I'll try to add that

**Comment by zimbatm** [comment]
Thanks, I've rebased, applied @fpletz's comments and added an alias for the rename.

**Comment by spinus** [comment]
Thanks

**Pull request closed without merging**